lib/rubocop/cop/naming/constant_name.rb

```diff
@@ -21,18 +21,32 @@ class ConstantName < Cop
         MSG = 'Use SCREAMING_SNAKE_CASE for constants.'.freeze
         SNAKE_CASE = /^[\dA-Z_]+$/
 
+        def_node_matcher :class_or_struct_return_method?, <<-PATTERN
+          (send
+            (const _ {:Class :Struct}) :new
+            ...)
+        PATTERN
+
         def on_casgn(node)
           _scope, const_name, value = *node
 
           # We cannot know the result of method calls like
           # NewClass = something_that_returns_a_class
           # It's also ok to assign a class constant another class constant
           # SomeClass = SomeOtherClass
-          return if value && (%i[block const casgn].include?(value.type) ||
-                    value.send_type? && value.method_name != :freeze)
+          return if value && %i[block const casgn].include?(value.type) ||
```

Comment:
You should update the comment preceding the changed code.

Reply:
I updated it. 092750fc3